### PR TITLE
TensorFlow CI fixes

### PR DESCRIPTION
**Pull request by Rocketknight1**
I made a lot of changes to the TF tests, and this exposed a few issues. This PR fixes all the exposed issues, so hopefully after this the only remaining CI issues should be related to generation or the `SharedEmbeddings` refactor.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

File: src/transformers/models/hubert/modeling_tf_hubert.py [hunk 1]
```diff
@@ -418,7 +418,8 @@ def build(self, input_shape):
         if not self.built:
             input_shape = input_shape.as_list()
             # Conv1D output shapes are checked at build time since TF 2.7, so we need to account for padding
-            input_shape[-2] += self.explicit_padding * 2
+            if input_shape[-2] is not None:
+                input_shape[-2] += self.explicit_padding * 2
```
Comment (ydshieh) [inline src/transformers/models/hubert/modeling_tf_hubert.py:422]: Could we add a short comment to explain why this condition? Even a link to a comment in one of your TF PR would be nice.

File: src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py [hunk 2]
```diff
@@ -452,7 +452,8 @@ def build(self, input_shape):
         if not self.built:
             input_shape = input_shape.as_list()
             # Conv1D output shapes are checked at build time since TF 2.7, so we need to account for padding
-            input_shape[-2] += self.explicit_padding * 2
+            if input_shape[-2] is not None:
+                input_shape[-2] += self.explicit_padding * 2
```
Comment (ydshieh) [inline src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py:456]: same

File: tests/models/blip/test_modeling_tf_blip.py [hunk 3]
```diff
@@ -434,6 +434,13 @@ def test_model_from_pretrained(self):
     def test_pt_tf_model_equivalence(self, allow_missing_keys=True):
         super().test_pt_tf_model_equivalence(allow_missing_keys=allow_missing_keys)
 
+    @unittest.skip("Matt: Re-enable this test when we have a proper export function for TF models.")
+    def test_saved_model_creation(self):
+        # This fails because the if return_loss: conditional can return None or a Tensor and TF hates that.
+        # We could fix that by setting the bool to a constant when exporting, but that requires a dedicated export
+        # function that we don't have yet.
```
Comment (ydshieh) [inline tests/models/blip/test_modeling_tf_blip.py:441]: I love this!

**Review by ydshieh — APPROVED** [review]
Thank you @Rocketknight1 ! Works for me, but maybe with one more comment would be nice to reader if there is ever someone want to look our code.

File: src/transformers/models/hubert/modeling_tf_hubert.py (hunk 1 above)
Comment (Rocketknight1) [reply src/transformers/models/hubert/modeling_tf_hubert.py:422]: Done!

File: src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py (hunk 2 above)
Comment (Rocketknight1) [reply src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py:457]: Done!

**Pull request merged**